### PR TITLE
Reduce Citus Disk Size

### LINKED ISSUE #9274: Citus disks in some envs are over-provisioned
### Problem

The default zfs disk size for workers is set to `3200GB` in our charts, however this is too big for some environments, for example, perfnet and previewnet. We should check the peak disk usage in these environments and determine a safe and conservative disk size.

### Solution

- Reduce the initial zfs disk size for those environments

### Alternatives

_No response_

### LINKED ISSUE #9276: POC Shrink Volume for Worker In Mainnet Environments
### Problem

The default in the chart for volume sizes is 3.2TB which is too big for the smallest worker with mainnet data.
We should shrink this disk to 1 TB. 

### Solution

1. Clone mainnet staging citus disks by following the create snapshots for citus runbook
2. Restore the disks to a test cluster by following the restoring citus from snapshot runbook
3. Once deployed access the smallest worker via postgres-util and get the minimum and maximum values for transaction.consensus_timestamp and document these values for later.
4. Scale down the citus nodepools to 0 
5. Create a new disk that is 1 TB
6. Create a vm (with enough space to store a snapshot) using ubuntu and install zfs utils
7. Mount the smaller workers disk and the new 1TB disk to vm
8. Take a zfs snapshot with `zfs snap`
9. Use zfs send to store on the vm filesystem
10. Use zfs restore to restore the snapshot to the 1TB disk
11. Take a snapshot of the 1 TB disk
12. Delete the original zfs disk and the new 1tb disk
13. restore the snapshot in step 11 to a new disk matching the name of the original zfs disk
14. Follow the restoring citus cluster from disk snapshots runbook
15. Verify data from step 3

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by jnels124**
**Description**:

* add script and runbook to reduce Citus disk size
* change default worker disk size in chart
* remove content encoding header for cached responses

**Related issue(s)**:

Fixes #9274 
Fixes #9276

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: docs/runbook/scripts/reduce-citus-disk-size.sh [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+kubectl delete pods -n "${COMMON_NAMESPACE}" -l 'app=zfs-init'
+sleep 5
+kubectl wait --for=condition=Ready pod -n "${COMMON_NAMESPACE}" -l 'app=zfs-init' --timeout=-1s
+
+log "Restarting ZFS node pods"
+kubectl delete pods -n "${COMMON_NAMESPACE}" -l 'component=openebs-zfs-node'
+sleep 5
+kubectl wait --for=condition=Ready pod -n "${COMMON_NAMESPACE}" -l 'component=openebs-zfs-node' --timeout=-1s
+
+NAMESPACES=($(echo $ZFS_VOLUMES | jq -r '.[].namespace' | tr ' ' '\n' | sort -u | tr '\n' ' '))
```
Comment (jnels124) [inline docs/runbook/scripts/reduce-citus-disk-size.sh:398]: will remove this line

File: docs/runbook/reduce-disksize-citus.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+## Problem
+
+Citus disks are over provisioned and need to be reduced in size.
+
+## Prerequisites
+
+- All Citus PVCS are defined are sized in Gi
```
Comment (steven-sheehy) [inline docs/runbook/reduce-disksize-citus.md:7]:
```suggestion
- All Citus PVCS are defined in GiB
```

File: docs/runbook/reduce-disksize-citus.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+- `jq` is installed
+- The kubectl context is set to the cluster you want to reduce disk size for from
```
Comment (steven-sheehy) [inline docs/runbook/reduce-disksize-citus.md:9]:
```suggestion
- The kubectl context is set to the cluster with the over-sized disk
```

File: docs/runbook/scripts/reduce-citus-disk-size.sh [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+for namespace in "${NAMESPACES[@]}"; do
+  unpauseCitus "${namespace}"
+  routeTraffic "${namespace}"
+done
```
Comment (steven-sheehy) [inline docs/runbook/scripts/reduce-citus-disk-size.sh:402]: Add a new line at EOF

File: docs/runbook/scripts/reduce-citus-disk-size.sh [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+updateK8sResources
+
+log "Restarting Stackgres pod"
+kubectl delete pods -n "${COMMON_NAMESPACE}" -l 'app=StackGresConfig'
+sleep 5
+kubectl wait --for=condition=Ready pod -n "${COMMON_NAMESPACE}" -l 'app=StackGresConfig' --timeout=-1s
+
+log "Restarting ZFS init daemonset"
+kubectl delete pods -n "${COMMON_NAMESPACE}" -l 'app=zfs-init'
+sleep 5
```
Comment (steven-sheehy) [inline docs/runbook/scripts/reduce-citus-disk-size.sh:390]: Why do we need the sleeps after deletes if we have the waits?

File: docs/runbook/scripts/reduce-citus-disk-size.sh [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+    "chroot /node-fs /bin/bash -c \
+    'zpool import ${POOL_NAME} && \
+     zpool upgrade ${POOL_NAME}'"
+    kubectl delete -n "${COMMON_NAMESPACE}" daemonset "${INSTANCE_NAME}"
+  done
+}
+
+function updateK8sResources() {
+  kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io \
+  "${HELM_RELEASE_NAME}" -o yaml > "/tmp/sg-webhook-${EPOCH_SECONDS}.yaml"
```
Comment (steven-sheehy) [inline docs/runbook/scripts/reduce-citus-disk-size.sh:275]: nit: Move "/tmp/sg-webhook-${EPOCH_SECONDS}.yaml" to a constant at top and reuse it below

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: docs/runbook/scripts/reduce-citus-disk-size.sh (hunk 5 above)
Comment (jnels124) [reply docs/runbook/scripts/reduce-citus-disk-size.sh:390]: Have had issues in the past where it doesn't see any pods with that label initially and the command just returns instantly with `error: no matching resources found`

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**